src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -44,21 +51,48 @@ const EditShiftHours = ({
 
   const inputName = 'shift';
   const { setServiceError, userId } = useApplicationContext();
+  const [isChecked, setIsChecked] = useState(false);
+  const [redirectTarget, setRedirectTarget] = useState('');
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const [clashingProperty, setClashingProperty] = useState(null);
+  const [clashingTimes, setClashingTimes] = useState(null);
+  const navigate = useNavigate();
+  const startEntryExists = !!timeEntry?.startTime && timeEntry.startTime !== '';
+  const finishEntryExists =
+    !!timeEntry?.finishTime && timeEntry.finishTime !== '';
+
   const {
     timeEntries,
     setTimeEntries,
     timecardDate,
     summaryErrors,
     setSummaryErrors,
+    setIsAlertVisible,
+    summaryMessages,
+    setSummaryMessages,
   } = useTimecardContext();
 
-  const [clashingProperty, setClashingProperty] = useState(null);
-  const [clashingTimes, setClashingTimes] = useState(null);
-
   useEffect(() => {
     focusErrors(document.getElementById('summary-error-0-message'));
+    if (shouldRedirect) {
+      navigate(redirectTarget);
```

Comment:
Are tests to cover this functionality to come in the near future?

Reply:
This is one of the focal points of next sprints spike

Reply:
cool, should we consider if we need to track this untested functionality somehow, e.g. a TODO comment or it written in a ticket somewhere before the PR gets merged...
Please feel free to resolve this.